ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -740,13 +779,40 @@ func (r *RayClusterReconciler) reconcilePods(ctx context.Context, instance *rayv
 		}
 		worker.ScaleStrategy.WorkersToDelete = []string{}
 
+		// Remove entire worker groups specified by MultihostGroupsToDelete to meet the expectations of the Autoscaler.
+		r.Log.Info("reconcilePods", "removing worker groups in the scaleStrategy of", worker.GroupName)
+		for _, groupToDelete := range worker.ScaleStrategy.MultihostGroupsToDelete {
+			for _, pod := range workerPods.Items {
+				if pod.Labels[utils.RayNodeGroupLabelKey] == groupToDelete {
```

Comment:
Thinking some more about this, I am not sure if this would work - if this is the worker group name, we'll end up deleting all pods from the worker group, including all the replicas. I think the intention here should be to delete only a specific replica, including all the hosts. 

Also I am not sure how the autoscaler logic would work. I am assuming the resource scheduler should know if certain pods have been idle, and it adds those pods to the deletion request. I don't think the autoscaler has any idea if an entire multihost replica has been idle.

@kevin85421 FYI

Reply:
After giving this a bit more thought, I think there are a couple of ways to solve this:

1. In the `get_node_data` function in Kuberay [node provider](https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/_private/kuberay/node_provider.py#L228), add some code to parse the replica index for each node.  
2. Then build a reverse lookup map that finds all other worker pods belonging to the same replica.
3. When one pod needs to scale down, make sure that all other pods from the same replica are also sent as part of the `WorkersToDelete`.

OR:

Similar to the above, but in step 3 we send the name of the replica to the scale down request. On the Kuberay operator, we figure out which exact pods belong to that replica and delete them together.